EpsilonAuxBmsV2/Core/Src/Tasks/OrionFunctions/Trip.c

```diff
@@ -1,20 +1,78 @@
 #include "Trip.h"
 
+
+void updateAuxTrip(OrionCanInfo* message, AuxTrip* auxTripToUpdate)
+{
+    uint8_t tripDuetoLowCell = 0;
+    uint8_t tripDuetoHighTempAndCurrent = 0;
+    uint8_t tripDuetoPackCurrent = 0;
+    uint8_t tripDuetoHighCell = 0;
+
+    const uint8_t chargeSense = !HAL_GPIO_ReadPin(CHARGE_SENSE_GPIO_Port, CHARGE_SENSE_Pin);
+    uint8_t protectionTrip = (!chargeSense && message->packCurrent < 0);
+
+
+    if (DEFAULT_VOLTAGE_UNITS * message->highCellVoltage >= ORION_MAX_CELL_VOLTAGE)
+    {
+        tripDuetoHighCell = 1;
+    }
+
+    // It's ok to just have a high temperature,
+    // as long as the car is not charging
+    if (message->highTemperature >= ORION_MAX_CHARGE_TEMP
+            && message->packCurrent < 0)
+    {
+        tripDuetoHighTempAndCurrent = 1;
+    }
+
+    // Charge current is negative
+    if (message->packCurrent <= ORION_MAX_CHARGE_CURRENT)
+    {
+        tripDuetoPackCurrent = 1;
+    }
+
+    if (DEFAULT_VOLTAGE_UNITS * message->lowCellVoltage <= ORION_MIN_CELL_VOLTAGE)
+    {
+        tripDuetoLowCell = 1;
+    }
+
+    // It's ok to just have a high temperature,
+    // as long as the car is not discharging
+    if (message->highTemperature >= ORION_MAX_DISCHARGE_TEMP
+            && message->packCurrent > 0)
+    {
+        tripDuetoHighTempAndCurrent = 1;
+    }
+
+    // Discharge current is positive
+    if (message->packCurrent >= ORION_MAX_DISCHARGE_CURRENT)
+    {
+        tripDuetoPackCurrent = 1;
+    }
+
+    auxTripToUpdate->protectionTrip = protectionTrip;
+    auxTripToUpdate->dischargeTripDueToLowCellVoltage = tripDuetoLowCell;
+    auxTripToUpdate->dischargeTripDueToHighTemperatureAndCurrent = tripDuetoHighTempAndCurrent;
+    auxTripToUpdate->dischargeTripDueToPackCurrent = tripDuetoPackCurrent;
+    auxTripToUpdate->chargeTripDueToHighCellVoltage = tripDuetoHighCell;
+    auxTripToUpdate->chargeTripDueToHighTemperatureAndCurrent = tripDuetoHighTempAndCurrent;
+    auxTripToUpdate->chargeTripDueToPackCurrent = tripDuetoPackCurrent;
+}
+
 /*
 Determines whether Discharge should cause a trip based on Orion CAN messages
 Discharge will cause a trip due to:
   * the min cell voltage being lower than the Orion limit
   * high temperature while discharging
   * discharge pack current being too high
-Sets aux trip variable
 */
-uint8_t checkDischargeTrip(OrionCanInfo* message, AuxTrip* auxTripToUpdate)
+uint8_t checkDischargeTrip(OrionCanInfo* message)
```

Comment:
Sort of confused by this one. The parameter is the message received from Orion via CAN. Maybe I'm just missing something tho..

Reply:
I was just thinking because a bunch of data is stored inside `localAuxTrip` to just use that instead of `message`, either works though.

Reply:
@duan-le you can choose what you want to do with this one

Reply:
I am also kind of confused on this one. So if we use `localAuxTrip` instead of `message`, we won't have access to
```
typedef struct
{
    uint16_t highCellVoltage;
    uint16_t lowCellVoltage;
    unsigned char highTemperature;
    float packCurrent;
} OrionCanInfo;
```
to do checks like
```
if (DEFAULT_VOLTAGE_UNITS * message->lowCellVoltage <= ORION_MIN_CELL_VOLTAGE)
{
        tripDuetoLowCell = 1;
}
```

Reply:
But doesn't `updateAuxTrip` already do all those checks already? It looks like a the stuff inside `checkDischargeTemp/checkChargeTemp` just repeats a lot of the code inside `updateAuxTrip`. I could be wrong, but I feel like the code could be simplified to:
```
updateAuxTrip(message, &localAuxTrip);
localAuxStatus.dischargeShouldTrip = checkDischargeTrip(localAuxStatus);

....
uint8_t checkDischargeTrip(localAuxStatus)
{
    return auxStatus.dischargeTripDueToLowCellVoltage ||
           auxStatus.dischargeTripDueToHighTemperatureAndCurrent ||
           auxStatus.dischargeTripDueToPackCurrent;
}
```
Similar to charge

Reply:
Oh I see what you mean now. And you are right, the check functions repeats what was done in `updateAuxTrip()`.